src/simplify_algebra.cpp

```diff
@@ -1008,40 +1021,58 @@ struct find_split_concat
 
     void apply(module& m, const match::matcher_result& r) const
     {
+        // Verifies that the slices meet several conditions: they must all output to the same
+        // concat instruction, slice on the same (1 only) axis, and the end of one slice
+        // must match the start of the next.
         auto ins = r.result;
 
         auto splits = get_splits(ins);
         if(splits.empty())
             return;
+        // Each slice must output to only one instruction
         if(std::any_of(
                splits.begin(), splits.end(), [](auto i) { return i->outputs().size() != 1; }))
             return;
-        // Check for concat operator
+        // The single output instruction for all items in the list must be the same one
         auto concat = splits.front()->outputs().front();
         if(std::any_of(splits.begin(), splits.end(), [&](auto i) {
                return i->outputs().front() != concat;
            }))
             return;
-        // Check axis match
+
+        // The axis for the common output instruction must be the same as for the split ops
         auto concat_op = any_cast<op::concat>(concat->get_operator());
         auto split_op  = any_cast<op::slice>(splits.front()->get_operator());
         if(split_op.axes.size() != 1)
             return;
         if(split_op.axes.front() != concat_op.axis)
             return;
-        // Replace args
+
+        // Filter the common instruction's inputs (concat can have any number of inputs)
+        // to the ones matching the split list
         auto args = concat->inputs();
         auto it =
             std::find_if(args.begin(), args.end(), [&](auto i) { return i == splits.front(); });
+        // Look for a series of slice operations equal to splits.size()
         if(std::distance(it, args.end()) < splits.size())
             return;
-        // If the slices are not in order then stop
+        // Don't do anything if the "slice" inputs to the concat op have other operations mixed in
+        // among them
+        if(std::any_of(it, it + splits.size(), [](instruction_ref x) {
+               return x->get_operator().name() != "slice";
+           }))
+            return;
+        // One slice must "start" where the last slice "end"  This parallels the check
+        // in get_splits(), ensuring the inputs to the concat op follow the same rule as the outputs
```

Comment:
This is not the same check that is done in `get_splits`. This is not checking that slice must "start" where the last slice "ends", which is already done in `get_splits`. This is checking that the slices passed to concat are in order.